src/dmd/clone.d

```diff
@@ -419,30 +413,25 @@ bool needOpEquals(StructDeclaration sd)
             if (ts.sym.isUnionDeclaration())
                 continue;
             if (needOpEquals(ts.sym))
-                goto Lneed;
+                return true;
             if (ts.sym.aliasthis) // https://issues.dlang.org/show_bug.cgi?id=14806
-                goto Lneed;
+                return true;
         }
         if (tv.isfloating())
         {
             // This is necessray for:
             //  1. comparison of +0.0 and -0.0 should be true.
             //  2. comparison of NANs should be false always.
-            goto Lneed;
+            return true;
         }
         if (tv.ty == Tarray)
-            goto Lneed;
+            return true;
         if (tv.ty == Taarray)
-            goto Lneed;
+            return true;
         if (tv.ty == Tclass)
-            goto Lneed;
+            return true;
     }
-Ldontneed:
-    //printf("\tdontneed\n");
```

Comment:
You see, this is why there's a common return.

Reply:
Wouldn't be be better to use an out contract if you want to examine the function exit?